.github/workflows/ci.yaml

```diff
@@ -0,0 +1,293 @@
+    runs-on: ubuntu-latest
+    if: needs.env_var.outputs.RunTestsOnly != 'true'
+    env:
+      AZURE_FUNCTIONAPP_PACKAGE_PATH: 'LoRaEngine/LoraKeysManagerFacade/'
+
+    steps:
+    - uses: actions/checkout@v2
+      name: Checkout current branch
+
+    - name: 'Resolve Project Dependencies Using Dotnet'
+      shell: bash
+      run: |
+        pushd './${{ env.AZURE_FUNCTIONAPP_PACKAGE_PATH }}'
+        dotnet build --configuration ${{ env.buildConfiguration }} --output ./output
+        popd
+
+    - name: 'Run Azure Functions Action'
+      uses: Azure/functions-action@v1
+      id: fa
+      with:
+        app-name: ${{ env.AZURE_FUNCTIONAPP_NAME }}
+        package: '${{ env.AZURE_FUNCTIONAPP_PACKAGE_PATH }}output'
+        publish-profile: ${{ secrets.AZURE_FUNCTIONAPP_PUBLISH_PROFILE }}
+
+  # Deploy IoT Edge solution to ARM gateway
+  deploy_arm_gw_iot_edge:
+    strategy:
+      fail-fast: false
+      max-parallel: 2
+      matrix: 
+        gateway: [ primary, secondary ]
+        include:
+          - gateway: primary
+            architecture: arm32v7
+          - gateway: secondary
+            architecture: arm32v7
+    needs: 
+    - env_var
+    - check_if_deploy
+    if: needs.env_var.outputs.RunTestsOnly != 'true'
+    env:
+      CONTAINER_REGISTRY_ADDRESS: "loramoduleintegrationtest.azurecr.io"
+      CONTAINER_REGISTRY_SERVER: "loramoduleintegrationtest.azurecr.io"
+      CONTAINER_REGISTRY_USERNAME: "loramoduleintegrationtest"
+      CONTAINER_REGISTRY_PASSWORD: ${{ secrets.ACR_CREDENTIALS }}
+      IOTHUB_CONNECTION_STRING: ${{ secrets.IOTHUB_CONNECTION_STRING }}
+      IOT_EDGE_DEPLOYMENT_ID: "citest${{ matrix.gateway }}"
+      NET_SRV_LOG_LEVEL: 1
+      NET_SRV_LOGTO_HUB: false
+      NET_SRV_LOGTO_UDP: true
+      NET_SRV_IOTEDGE_TIMEOUT: 0
+      NET_SRV_LOG_TO_UDP_ADDRESS: "192.168.1.2"
+      EDGEHUB_ROUTE: "FROM /* INTO $upstream"
+      RESET_PIN: 7
+      REGION: "EU"
+      FACADE_SERVER_URL: "https://${{ needs.env_var.outputs.AZURE_FUNCTIONAPP_NAME }}.azurewebsites.net/api/"
+      FACADE_AUTH_CODE: ${{ secrets.FUNCTION_AUTH_CODE }}
+      EDGE_AGENT_VERSION: 1.0.10
+      EDGE_HUB_VERSION: 1.0.10
+      EDGEHUB_OPTIMIZEFORPERFORMANCE: false
+      EDGEHUB_MQTTSETTINGS_ENABLED: false
+      EDGEHUB_HTTPSETTINGS_ENABLED: false
+      BUILD_BUILDID: ${{ github.run_id }}
+      DEV_IMAGE_TAG: dev
+      DEVICE_ID: ${{ matrix.gateway }}
+      MASTER_IMAGE_TAG: master
+    runs-on: ubuntu-latest
+    name: Run IoT Edge Deployment on ARM gateway
+    steps:  
+      - uses: actions/checkout@v2
+        name: Checkout current branch
+
+      - run: |
+          if [ ${{ github.ref	}} = "refs/heads/dev" ]; then
+          echo "dev"  
+          IMAGE_TAG="$DEV_IMAGE_TAG"
+          elif [ ${{ github.ref	}} = "refs/heads/master" ]; then
+          echo "master"
+          IMAGE_TAG="$MASTER_IMAGE_TAG"
+          elif [ ${{ github.event_name }}	== "pull_request" ]; then
```

Comment:
Is there a reason for the odd spacing?

Reply:
I don't know, locally it's not like this